polar2grid/writers/hdf5.py

```diff
@@ -0,0 +1,293 @@
+                mode = "w"
+        else:
+            LOG.info("Creating HDF5 file: %s", output_filename)
+            mode = "w"
+
+        h5_fh = h5py.File(output_filename, mode)
+        return h5_fh
+
+    @staticmethod
+    def create_proj_group(filename: str, parent: TextIO, area_def):
+        """Create the top level group from projection information."""
+        projection_name = (area_def.name).replace(" ", "_")
+        # if top group alrady made, return.
+        if projection_name in parent:
+            return projection_name
+
+        # create top group for first time
+        group = parent.create_group(projection_name)
+        # add attributes from grid_defintion.
+        if isinstance(area_def, SwathDefinition):
+            group.attrs["height"], group.attrs["width"] = area_def.shape
+            group.attrs["description"] = "No projection: native format"
+        else:
+            for a in ["height", "width", "origin_x", "origin_y"]:
+                ds_attr = getattr(area_def, a, None)
+                if ds_attr is None:
+                    pass
+                else:
+                    group.attrs[a] = ds_attr
+            group.attrs["proj4_string"] = area_def.proj_str
+            group.attrs["cell_height"] = area_def.pixel_size_y
+            group.attrs["cell_width"] = area_def.pixel_size_x
+
+        return projection_name
+
+    @staticmethod
+    def check_variable(fh, var_name: str, data: da.array, dtype: np.dtype, append: bool, compression: bool):
+        if append:
+            if var_name in fh:
+                LOG.warning("Product %s already exists in hdf5 group, will delete existing dataset", var_name)
+                del fh[var_name]
+        dset = fh.create_dataset(var_name, shape=data.shape, dtype=dtype, compression=compression)
+
+    def write_geolocation(
+        self, fh, fname: str, parent: str, area_def, dtype: np.dtype, append: bool, compression, chunks: tuple[int, int]
+    ) -> tuple[list, list[FakeHDF5]]:
+        """Delayed Geolocation Data write."""
+        msg = ("Adding geolocation 'longitude' and " "'latitude' datasets for grid %s", parent)
+        LOG.info(msg)
+        lon_data, lat_data = area_def.get_lonlats(chunks=chunks)
+
+        dtype = lon_data.dtype if dtype is None else dtype
+
+        lon_grp = "{}/longitude".format(parent)
+        lat_grp = "{}/latitude".format(parent)
+
+        lon_dataset = FakeHDF5(fname, lon_grp)
+        lon_dset = self.check_variable(fh, lon_grp, lon_data, dtype, append, compression)
+        lat_dataset = FakeHDF5(fname, lat_grp)
+        lat_dset = self.check_variable(fh, lat_grp, lat_data, dtype, append, compression)
+
+        return [lon_data, lat_data], [lon_dataset, lat_dataset]
+
+    @staticmethod
+    def create_variable(filename: str, parent, hdf_subgroup: str, dataset_id: xr.DataArray, dtype: np.dtype, **kwargs):
+        """Create a hdf5 subgroup and it's attributes."""
+
```

Comment:
Can you remove this blank line? Kind of surprised black doesn't catch this either.